yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,264 @@
+            'uploader_id': '62b8bdfc9021052f7905882d',
+            'view_count': int,
+        }
+    }, {
+        'url': 'https://allstar.gg/montage?montage=RILJMH6QOS',
+        'info_dict': {
+            'id': '64a2697372ce3703de29e868',
+            'display_id': 'RILJMH6QOS',
+            'title': 'cherokee Rapid Fire Snipers Montage',
+            'url': 'md5:d5672e6f88579730c2310a80fdbc4030',
+            'thumbnail': 'md5:60872f0d236863bb9a6f3dff1623403c',
+            'uploader': 'cherokee',
+            'ext': 'mp4',
+            'timestamp': 1688365434271,
+            'uploader_id': '62b8bdfc9021052f7905882d',
+            'view_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        query_id, video_id = self._match_valid_url(url).groups()
+
+        assert query_id in _QUERIES
+
+        return _parse_video_data(
+            self._send_query(
+                _QUERIES.get(query_id), {'id': video_id},
+                ('data', 'video'), video_id))
+
+
+class AllstarProfileIE(AllstarBase):
+    _VALID_URL = r'https?://(?:www\.)?allstar\.gg/(?:profile\?user=|u/)(?P<id>[^/?#&]+)'
+
+    _TESTS = [{
+        'url': 'https://allstar.gg/profile?user=62b8bdfc9021052f7905882d',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Clips',
+        },
+        'playlist_mincount': 15
+    }, {
+        'url': 'https://allstar.gg/u/cherokee?game=730&view=Clips',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Clips',
+        },
+        'playlist_mincount': 15
+    }, {
+        'url': 'https://allstar.gg/u/62b8bdfc9021052f7905882d?view=Montages',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Montages',
+        },
+        'playlist_mincount': 4
+    }, {
+        'url': 'https://allstar.gg/profile?user=cherokee&view=Mobile Clips',
+        'info_dict': {
+            'id': '62b8bdfc9021052f7905882d',
+            'title': 'cherokee - Mobile Clips',
+        },
+        'playlist_mincount': 1
+    }]
+
+    def _get_page(self, user_id, game, query_id, page_num):
+        page_num += 1
+
+        for video_data in self._send_query(
+                _QUERIES.get(query_id), {
+                    'user': user_id,
+                    'page': page_num,
+                    'game': int_or_none(game),
+                }, ('data', 'videos', 'data'), f'{user_id} page {page_num}'):
+            yield _set_webpage_url(_parse_video_data(video_data))
+
+    def _get_user_data(self, user_id, path=()):
+        return traverse_obj(
+            self._download_json(
+                urljoin('https://api.allstar.gg/v1/users/profile/', user_id),
+                user_id), path)
+
+    def _real_extract(self, url):
+        user_id = self._match_id(url)
+        user_id, user_name = self._get_user_data(
+            user_id, ('data', ('_id', ('profile', 'username'))))
```

Comment:
Sorry I don't understand what you mean here, can you explain more/paraphrase please.

Reply:
Okay, say the website ever changes their data structure and instead of `username` the value is now stored at `profilename`. The `traverse_obj` call inside `_get_user_data` would then only return a single value, which could not be unpacked and assigned to `user_id, user_name`. So even though only the username could not be extracted, the whole extractor is now failing.

Since `_get_user_data` isn't reused anyway, maybe this could be used:
```python
        display_id = self._match_id(url)
        profile_data = self._download_json(
            urljoin('https://api.allstar.gg/v1/users/profile/', display_id), display_id)['data']

        user_id = profile_data['_id']
        username = traverse_obj(profile_data, ('profile', 'username'))
```

Reply:
ok I get it

Reply:
take a look and tell me if it is good now.